java/common/common-task/src/main/java/sleeper/common/task/RunDataProcessingTasks.java

```diff
@@ -53,28 +57,63 @@
  * Finds the number of messages on a queue, and starts up one EC2 or Fargate task for each, up to a
  * configurable maximum.
  */
-public class RunCompactionTasks {
-    private static final Logger LOGGER = LoggerFactory.getLogger(RunCompactionTasks.class);
+public class RunDataProcessingTasks {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RunDataProcessingTasks.class);
 
     private final InstanceProperties instanceProperties;
     private final TaskCounts taskCounts;
     private final CompactionTaskHostScaler hostScaler;
     private final TaskLauncher taskLauncher;
+    private CompactionECSLaunchType launchType;
+    private String containerName;
+    private String clusterName;
+    private String fargateDefinitionFamily;
 
-    public RunCompactionTasks(
-            InstanceProperties instanceProperties, EcsClient ecsClient, AutoScalingClient asClient, Ec2Client ec2Client) {
-        this(instanceProperties,
-                () -> ECSTaskCount.getNumPendingAndRunningTasks(instanceProperties.get(COMPACTION_CLUSTER), ecsClient),
-                EC2Scaler.create(instanceProperties, asClient, ec2Client),
-                (numberOfTasks, checkAbort) -> launchTasks(ecsClient, instanceProperties, numberOfTasks, checkAbort));
+    public RunDataProcessingTasks(
+            InstanceProperties instanceProperties,
+            EcsClient ecsClient,
+            AutoScalingClient asClient,
+            Ec2Client ec2Client,
+            CompactionTaskType compactionTaskType) {
+
+        this.instanceProperties = instanceProperties;
+
+        // Initialize taskCounts lambda
+        this.taskCounts = () -> ECSTaskCount.getNumPendingAndRunningTasks(
+                instanceProperties.get(compactionTaskType == CompactionTaskType.BULK_EXPORT ? BULK_EXPORT_CLUSTER : COMPACTION_CLUSTER),
+                ecsClient);
+
+        // Initialize hostScaler
+        this.hostScaler = EC2Scaler.create(instanceProperties, asClient, ec2Client);
+
+        // Set taskLauncher to call internal method
+        this.taskLauncher = (numberOfTasks, checkAbort) -> this.launchTasks(ecsClient, instanceProperties, numberOfTasks, checkAbort);
+
+        // Determine launch type and related configs
+        if (compactionTaskType == CompactionTaskType.BULK_EXPORT) {
```

Comment:
Rather than an enum and all this work in the constructor, you could have a private constructor that takes all the parameters, including the ones that depend on the task type. You could then have two public static methods that call the constructor, one for compaction and one for bulk export.

That way the dependencies of the class are declared explicitly in one place, and the way in which they're configured for the different task types are declared together in one place for each task type.